msteams-platform/bots/how-to/Teams conversational AI/how-conversation-ai-core-capabilities.md

```diff
@@ -285,73 +285,112 @@ A simple interface for actions and predictions allows bots to react when they ha
+    Examples:  
 
-Example: [List bot](https://github.com/microsoft/teams-ai/tree/main/js/samples/04.ai.d.chainedActions.listBot)
+    Human: remind me to buy milk
+    AI: DO addItem list="groceries" item="milk" THEN SAY Ok I added milk to your groceries list
+    Human: we already have milk
+    AI: DO removeItem list="groceries" item="milk" THEN SAY Ok I removed milk from your groceries list
+    Human: buy ingredients to make margaritas
+    AI: DO addItem list="groceries" item="tequila" THEN DO addItem list="groceries" item="orange liqueur" THEN DO addItem list="groceries" item="lime juice" THEN SAY Ok I added tequila, orange liqueur, and lime juice to your groceries list
+    Human: do we have have milk
+    AI: DO findItem list="groceries" item="milk"
+    Human: what's in my grocery list
+    AI: DO summarizeLists  
+    Human: what's the contents of all my lists?
+    AI: DO summarizeLists
+    Human: show me all lists but change the title to Beach Party
+    AI: DO summarizeLists
+    Human: show me all lists as a card and sort the lists alphabetically
+    AI: DO summarizeLists
 
-[Sample code reference](https://github.com/microsoft/teams-ai/blob/main/js/samples/04.ai.d.chainedActions.listBot/src/index.ts#L149)
+    ```
 
-```typescript
-app.ai.action('addItem', async (context, state, data: EntityData) => {
-    const items = getItems(state, data.list);
-    items.push(data.item);
-    setItems(state, data.list, items);
-    return true;
-});
+* Conversation history:
 
+    ```
+    {{conversation.history}} 
+    ```
 
-app.ai.action('removeItem', async (context, state, data: EntityData) => {
-    const items = getItems(state, data.list);
-    const index = items.indexOf(data.item);
-    if (index >= 0) {
-        items.splice(index, 1);
-        setItems(state, data.list, items);
-        return true;
-    } else {
-        await context.sendActivity(responses.itemNotFound(data.list, data.item));
+* Current query:
 
+    ```
+    Human: {{activity.text}} 
+    ```
 
-        // End the current chain
-        return false;
-    }
-});
+* Current list names:
 
-app.ai.action('findItem', async (context, state, data: EntityData) => {
-    const items = getItems(state, data.list);
-    const index = items.indexOf(data.item);
-    if (index >= 0) {
-        await context.sendActivity(responses.itemFound(data.list, data.item));
-    } else {
-        await context.sendActivity(responses.itemNotFound(data.list, data.item));
-    }
+    ```javascript
+    {{conversation.listNames}}
+    ```
 
+* AI: The bot logic is simplified to provide handlers for actions such as `addItem`, `removeItem`, and `findItem`. This clear delineation between actions and the prompts that instruct the AI on how to execute them is an incredibly potent tool.
```

Comment:
The bot logic is streamlined to include handlers for actions such as `addItem`, `removeItem`, and `findItem`. This distinct separation between actions and the prompts guiding the AI on their execution serves as a powerful tool.

Reply:
Updated.